packages/next-auth/src/providers/strava.ts

```diff
@@ -0,0 +1,60 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface StravaProfile extends Record<string, any> {
+  id: string // this is really a number
+  username: string
+  resource_state: number
+  firstname: string
+  lastname: string
+  bio: string
+  city: string
+  state?: any
+  country?: any
+  sex: string
+  premium: boolean
+  summit: boolean
+  created_at: string
+  updated_at: string
+  badge_type_id: number
+  weight: number
+  profile_medium: string
+  profile: string
+  friend?: any
+  follower?: any
+}
+
+export default function Strava<P extends StravaProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "strava",
+    name: "Strava",
+    type: "oauth",
+    authorization: {
+      url: "https://www.strava.com/api/v3/oauth/authorize",
+      params: {
+        scope: "read",
+        approval_prompt: "auto",
+        response_type: "code",
+        redirect_uri: `${process.env.NEXTAUTH_URL|| `https:localhost:3000`}/api/auth/callback/strava`,
```

Comment:
```suggestion
      params: {
        scope: "read",
        approval_prompt: "auto",
        response_type: "code",
```
NextAuth by default will use `{host}/api/auth/callback/strava` as the redirect URL when calling the provider, no need to do anything here ☕️

Reply:
For my own learning, where does the `/api/auth/callback/strava` get picked up in the `redirect_uri` construction? Can you point me to it? Happy to remove the current, code, though. 
Does the filename `strava` act as the provider identifier?

Reply:
@michaelangrivera you can see here in the code:

https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/lib/providers.ts#L25-L30

How when parsing a provider, we do a deep merge of the supplied options with the default options. For the callback URL, we use the ID  to create the path: `/api/auth/callback/{id}`.

For the domain, the URL will be `NEXTAUTH_URL` if set, if not `http://localhost:3000/api/auth`.

Reply:
There's also no need to set `redirect_uri` explicitly on the provider config; this is done automatically by **NextAuth** when issuing the OAuth request:

https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/lib/oauth/client.ts#L39

Reply:
Great! Thank you!